src/data-source.js

```diff
@@ -61,11 +61,10 @@ export default class DataSource {
         return callback(err);
       }
       if (objects[0]) {
-        if (thisIdx > 0) {
+        if (thisIdx >= 0) {
           return this.process(this.sources[thisIdx - 1], method, params, objects, callback);
```

Comment:
`thisIdx - 1` looks like a bad idea when thisIdx is 0.

Reply:
as crazy as it sounds...there is a check in `this.process` that _expects_ the data source to be null, and seems to be how it "closes" the loop on data sources

Reply:
i can change it so that the invocation is more explicit though, since i think that would make it easier to understand